compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FakeInliningLocalVariablesLowering.kt

```diff
@@ -0,0 +1,88 @@
+import org.jetbrains.kotlin.ir.util.parentAsClass
+import org.jetbrains.kotlin.ir.visitors.acceptChildrenVoid
+import org.jetbrains.kotlin.ir.visitors.acceptVoid
+import org.jetbrains.kotlin.load.java.JvmAbi
+
+internal val fakeInliningLocalVariablesLowering = makeIrFilePhase(
+    ::FakeInliningLocalVariablesLowering,
+    name = "FakeInliningLocalVariablesLowering",
+    description = "Add fake locals to identify the range of inlined functions and lambdas"
+)
+
+internal class FakeInliningLocalVariablesLowering(val context: JvmBackendContext) : IrInlineReferenceLocator(context), FileLoweringPass {
+    override fun lower(irFile: IrFile) {
+        irFile.acceptVoid(this)
+    }
+
+    override fun visitFunctionNew(declaration: IrFunction) {
+        declaration.acceptChildrenVoid(this)
+        if (declaration.isInline && !declaration.origin.isSynthetic && declaration.body != null) {
+            declaration.addFakeInliningLocalVariables()
+        }
+    }
+
+    override fun handleInlineFunctionCallableReferenceParam(valueArgument: IrCallableReference) {
+        // Do not record inline function callable reference parameters. They will not be used.
+    }
+
+    override fun handleInlineFunctionLambdaParam(lambda: IrFunction, callee: IrFunction, callSite: IrDeclaration?) {
+        // Do not record lambda parameters. Instead deal with them now.
+        if (lambda.origin == IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA) {
+            lambda.addFakeInliningLocalVariablesForArguments(callee)
+        }
+    }
+
+    private fun IrFunction.addFakeInliningLocalVariables() {
+        val currentFunctionName = context.methodSignatureMapper.mapFunctionName(this)
+        val localName = "${JvmAbi.LOCAL_VARIABLE_NAME_PREFIX_INLINE_FUNCTION}$currentFunctionName"
+        addFakeLocalVariable(localName)
+    }
+
+    private fun IrFunction.addFakeInliningLocalVariablesForArguments(callee: IrFunction) {
+        val currentFunctionName = context.methodSignatureMapper.mapFunctionName(this)
+        val argumentToFunctionName = context.methodSignatureMapper.mapFunctionName(callee)
+        val internalName = context.typeMapper.mapClass(parentAsClass).internalName
+        val thisType = internalName.substringAfterLast('/', internalName)
+        val lambdaReference = "$thisType\$$currentFunctionName"
+        val localName = "${JvmAbi.LOCAL_VARIABLE_NAME_PREFIX_INLINE_ARGUMENT}-$argumentToFunctionName-$lambdaReference"
+        addFakeLocalVariable(localName)
+    }
+
+    private fun IrFunction.addFakeLocalVariable(name: String) {
+        context.createIrBuilder(symbol).run {
+            body = irBlockBody {
+                // Create temporary variable, but make sure it's origin is `DEFINED` so that
+                // it will materialize in the code.
+                createTmpVariable(irInt(0), name, origin = IrDeclarationOrigin.DEFINED)
+                if (body is IrExpressionBody) {
+                    +(body as IrExpressionBody).expression
```

Comment:
@madsager Last monent  thought: seems this branch is not executed at all cause there should be additional return

Reply:
Nice catch, I have the same error in bridge lowering. I'll see if I can generate test cases hitting this.